kythe/cxx/indexer/cxx/IndexerASTHooks.cc

```diff
@@ -1049,11 +1049,72 @@ bool IndexerASTVisitor::TraverseCXXConstructorDecl(
          TraverseDeclarationNameInfo(DNI);
 }
 
+bool IndexerASTVisitor::ShouldIndex(const clang::Decl* Decl) {
+  if (TemplateInstanceExcludePathPattern == nullptr) {
+    return true;
+  }
```

Comment:
This is a lot of code, that conceptually is simple(-ish). I think it's worth a comment here describing that we return true if the decl is an explicit instantiation or a (partial) specialization, or if the regexp does not match the containing source file path.